server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -564,11 +681,485 @@ protected Point creteInfluxDbPoint(Object metricsObject) {
         }
     }
 
+     class DbCollector extends AbstractStatsCollector {
+         List<Double> loadHistory = new ArrayList<>();
+         DbCollector() {
+             dbStats.put(loadAvarages, loadHistory);
+         }
+         @Override
+         protected void runInContext() {
+             LOGGER.debug(String.format("%s is running...", this.getClass().getSimpleName()));
+
+             try {
+                 int lastUptime = (dbStats.containsKey(uptime) ? (Integer) dbStats.get(uptime) : 0);
+                 int lastQueries = (dbStats.containsKey(queries) ? (Integer) dbStats.get(queries) : 0);
+                 getDynamicDataFromDB();
+                 int interval = (Integer) dbStats.get(uptime) - lastUptime;
+                 int activity = (Integer) dbStats.get(queries) - lastQueries;
+                 loadHistory.add(0, Double.valueOf(activity / interval));
+                 int maxsize = DATABASE_SERVER_LOAD_HISTORY_RETENTION_NUMBER.value();
+                 while (loadHistory.size() > maxsize) {
+                     loadHistory.remove(maxsize - 1);
+                 }
+             } catch (Throwable e) {
+                 // pokemon catch to make sure the thread stays running
+                 LOGGER.error("db statistics collection failed due to " + e.getLocalizedMessage());
+                 if (LOGGER.isDebugEnabled()) {
+                     LOGGER.debug("db statistics collection failed.", e);
+                 }
```

Comment:
Do we need 2 logs?

Reply:
I did this on purpose to only add the stack on debug level and not at `error`.